major-mode-hydra.el

```diff
@@ -134,10 +134,14 @@ exactly the same structure as that in `pretty-hydra-define' or
 
 (defun major-mode-hydra-dispatch (mode)
   "Summon the hydra for given MODE (if there is one)."
-  (let ((hydra (major-mode-hydra--body-name-for mode)))
-    (if (fboundp hydra)
-        (call-interactively hydra)
-      (message "Major mode hydra not found for %s" mode))))
+  (catch 'done
+    (while mode
+      (let ((hydra (major-mode-hydra--body-name-for mode)))
+        (when (fboundp hydra)
+          (call-interactively hydra)
+          (throw 'done t)))
+      (setq mode (get-mode-local-parent mode)))
```

Comment:
Is `get-mode-local-parent` a built-in function? I got `Symbol's function definition is void` when trying it on emacs 27.

Reply:
`get-mode-local-parent` is defined in `mode-local.el`, but the function is not autoloaded. It is defined as follows:

```
(defsubst get-mode-local-parent (mode)
  "Return the mode parent of the major mode MODE.
Return nil if MODE has no parent."
  (or (get mode 'mode-local-parent)
      (get mode 'derived-mode-parent)))
```

It is possible to load the function explicitly, but `(get mode 'derived-mode-parent)` would be usually sufficient, so I want to replace the expression in that way.  What do you think?

Reply:
Yeah I think `(get mode 'derived-mode-parent)` should be good enough.